src/services/persistent-storage.service.ts

```diff
@@ -21,36 +53,42 @@ export class PersistentStorage {
   private init() {
     for (let i = 0; i < this.storage.length; i++) {
       const key = this.storage.key(i);
-      if (key?.startsWith(`${this.prefix}-`)) {
+      if (key?.startsWith(`${this.prefix}:`)) {
         const value = this.storage.getItem(key);
         if (value) {
-          this.map.set(key.replace(`${this.prefix}-`, ""), value);
+          this.map.set(key.replace(`${this.prefix}:`, ""), value);
         }
       }
     }
   }
 
   clear(): void {
     this.map.clear();
+    this.storage.clear();
   }
+
   getItem(key: string): string | null {
```

Comment:
We'll need to change the return type for this method now as it won't always be a `string` or `null`. I think we should support something similar to what we have with `.contract.view()` where we can pass a return type:
```ts
// Example Usage:
storage.getItem<AuthData>(LOCAL_STORAGE_LEDGER_WALLET_AUTH_DATA);

// Implementation:
getItem<Value extends unknown>(key: string): Value | null {
  const value = this.map.get(key);

  return typeof value === "undefined"
    ? JSON.parse(this.map.get(key))
    : null;
}

```
This will remove the "@ts-ignore" comment(s) too